python/ray/_private/runtime_env/__init__.py

```diff
@@ -1,3 +1,48 @@
+import json
+import os
+import sys
+from typing import Any, Dict
+
+
+def runtime_env_override_hook(runtime_env: Dict[str, Any]) -> Dict[str, Any]:
+    """Hook to override the runtime environment from RAY_OVERRIDE_RUNTIME_ENV_JSON."""
+
+    if "RAY_OVERRIDE_RUNTIME_ENV_JSON" in os.environ:
+        runtime_env = json.loads(os.environ["RAY_OVERRIDE_RUNTIME_ENV_JSON"])
+
+    return runtime_env
+
+
+def uv_run_runtime_env_hook(runtime_env: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
now where do we inject this hook? in snapshot_util? 

tangent question: How do we deal with uv run on jobs?

Reply:
It is actually very simple, you just submit the job with `uv run <script>` as the entry point :)

On the hook: Right now you need to define it manually but I think going forward we should just remove the hook and do this by default once it is ironed out a little more and works well :)

Reply:
so we still need to define the env variable `RUNTIME_ENV_HOOK` (I forget its name)?

Reply:
Yes, that's right (for now) :)